framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -149,304 +162,209 @@ public boolean hasDependentAnnotations() {
         }
         List<String> elements = new ArrayList<>();
         for (Method method : methods) {
-            org.checkerframework.framework.qual.JavaExpression javaExpression =
+            org.checkerframework.framework.qual.JavaExpression javaExpressionAnno =
                     method.getAnnotation(org.checkerframework.framework.qual.JavaExpression.class);
-            if (javaExpression != null) {
+            if (javaExpressionAnno != null) {
                 elements.add(method.getName());
             }
         }
         return elements;
     }
 
-    ///
-    /// Viewpoint adaptation
-    ///
+    /**
+     * Returns the elements of the annotation that are Java expressions.
+     *
+     * @param am AnnotationMirror
+     * @return the elements of the annotation that are Java expressions
+     */
+    private List<String> getListOfExpressionElements(AnnotationMirror am) {
+        for (Class<? extends Annotation> clazz : annoToElements.keySet()) {
```

Comment:
Why isn't this `annoToElements.get(am.getClass())`?  `Class` objects are interned.  Is it because the AnnotationMirror might be built by javac or by the Checker Framework, or for some other reason?  Another possibility is to have two maps, one with Class as the key and one with String as the key.

There are other uses of this map that could also perhaps be optimized.

Reply:
`am.getClass()` returns `AnnotationMirror`.  `AnnotationMirror` doesn't have a field that stores the `Class` of the annotation it represents.  The key of this map should be the annotation's name (as a String) rather than a class.  We are also going to refactor this map so that the values are elements after this pull request is complete.  There's a todo at the declaration of this field.